### ISSUE TITLE
Support for more fields?

### ISSUE THREAD

**Issue by Mikec78660** [oskvr37/tiddl#40]
Will you be adding all the fields from tidal-dl? Like RecordType, Release date, year, etc. for example? I was hoping to have this as a drop in replacement but it doesn't seem like you cannot create the same folder structure currently so it will not detect if files already exist. Or am I missing something?

**Comment by oskvr37** [comment]
Do you mean [this](https://github.com/yaronzz/Tidal-Media-Downloader?tab=readme-ov-file#settings---possible-tags)?
As I remember I did not add more fields as it had to make additional request for more details about track (year etc.)
But I can try to add this feature

**Comment by Mikec78660** [comment]
Yes exactly that! Oh it isn't part of the json that is returned from the get album call? If it has to make additional requests maybe that wouldn't be a good thing. would it cause any type of rate limit issues with Tidal?

{
  "id": 38519994,
  "title": "Messier Objects",
  "duration": 3247,
  "streamReady": true,
  "streamStartDate": "2015-01-30T00:00:00.000+0000",
  "allowStreaming": true,
  "premiumStreamingOnly": false,
  "numberOfTracks": 17,
  "numberOfVideos": 0,
  "numberOfVolumes": 1,
  "releaseDate": "2015-02-03",
  "copyright": "Alien Transistor",
  "type": "ALBUM",
  "version": null,
  "url": "http://www.tidal.com/album/38519994",
  "cover": "44226420-6f7e-45d0-9dc6-197c4f4a5cb1",
  "videoCover": null,
  "explicit": false,
  "upc": "880918220925",
  "popularity": 19,
  "audioQuality": "LOSSLESS",
  "audioModes": [
    "STEREO"
  ],
  "artist": {
    "id": 3529689,
    "name": "The Notwist",
    "type": "MAIN"
  },
  "artists": [
    {
      "id": 3529689,
      "name": "The Notwist",
      "type": "MAIN"
    }
  ]
}

**Comment by papavlos** [comment]
@oskvr37 Please note that you already capture some more fields when constructing metadata for ID4.
See function "setMetadata" in source file "lib/python3.12/site-packages/tiddl/utils.py".

I have easy defined two additional fields "date" and "discnumber" by inserting two lines after the line 53 in the same file in function "formatFilename":
`        "date": track["streamStartDate"][:10].strip(),`
`        "discnumber": str(track["volumeNumber"]).strip(),`

Now I can use them to output tracks like this:
` -o "{artist} - ({date}) {album}/{discnumber}.{number}.{title}"`

It would be nice if you use it within future relesae ;)

**Comment by oskvr37** [comment]
I added new fields
- disc_number
- released
- year
Please leave me a feedback about this change, should I add something more or tweak these fields
